From 147b027b512c362bcf65ef73314106a727f09847 Mon Sep 17 00:00:00 2001
From: Robert Church <robert.church@windriver.com>
Date: Wed, 11 May 2016 00:06:05 -0500
Subject: [PATCH 1/7] Pike Rebase: Adding support of Keyring

commit bac23c20571f365d8912a5b6c64d041403ffdfee
Author: Sylvain Chenard <sylvain.chenard@windriver.com>
Date:   Tue Sep 23 11:01:35 2014 -0400

    Adding support of Keyring in OCF script,
    creation of a binary to suppress cleartext password
    in logs

Related:
 - Newton commit: (CGCS_DEV_0024) 93124fe
 - Mitaka commits: (CGCS_DEV_0019) e817180
                   (CGCS_DEV_0019_squash) e817180
 - Kilo commits: (CGCS_DEV_0016) b5b4176

(cherry picked from commit e8171807dd545d6a05f4ccb22dab41881c5179ba)
Signed-off-by: Robert Church <robert.church@windriver.com>
---
 glanceclient/shell.py | 8 ++++++++
 requirements.txt      | 1 +
 2 files changed, 9 insertions(+)

diff --git a/glanceclient/shell.py b/glanceclient/shell.py
index 416d173..6b0672e 100644
--- a/glanceclient/shell.py
+++ b/glanceclient/shell.py
@@ -29,6 +29,8 @@ import os
 import sys
 import traceback
 
+import keyring
+
 from oslo_utils import encodeutils
 from oslo_utils import importutils
 import six
@@ -359,6 +361,12 @@ class OpenStackImagesShell(object):
                   "env[OS_USERNAME]"))
 
         if not args.os_password:
+            # priviledge check (only allow Keyring retrieval if we are root)
+            if os.geteuid() == 0:
+                args.os_password = keyring.get_password('CGCS',
+                                                        args.os_username)
+
+        if not args.os_password:
             # No password, If we've got a tty, try prompting for it
             if hasattr(sys.stdin, 'isatty') and sys.stdin.isatty():
                 # Check for Ctl-D
diff --git a/requirements.txt b/requirements.txt
index dcdc24f..90232c8 100644
--- a/requirements.txt
+++ b/requirements.txt
@@ -12,3 +12,4 @@ oslo.utils>=3.20.0 # Apache-2.0
 oslo.i18n!=3.15.2,>=2.1.0 # Apache-2.0
 wrapt>=1.7.0 # BSD License
 pyOpenSSL>=0.14 # Apache-2.0
+keyring>=5.3
-- 
2.7.4

